snapcraft.yaml

```diff
@@ -84,10 +84,23 @@ parts:
       - bin/microcloud
       - bin/microcloudd
 
+  vim:
+      plugin: nil
+      stage-packages:
+        - vim-tiny
+      override-build: |
+        touch "${CRAFT_PART_INSTALL}/usr/share/vim/vim91/defaults.vim"
+      organize:
+        usr/bin: bin/
```

Comment:
Back when LXD was bundling `vim.tiny`, we also had this organize line:
```
usr/share/vim/vim*/debian.vim: etc/vimrc
```

Dunno if that would prevent to try and load the `defaults.vim` file that was never primed in the LXD case.

Reply:
I made some tests based on what I found in https://github.com/canonical/lxd-pkg-snap/pull/237. 
When you don't set `--clean`, then VIM tries to go through the entire chain of directories to find a config file. There is `-V` which gives you a nice rundown of probed files/dirs:

```
root@micro01:~# vim.tiny -V -Z /tmp/test
chdir(/tmp)
fchdir() to previous dir
chdir(/etc/vim)
fchdir() to previous dir
sourcing "/etc/vim/vimrc"
chdir(/usr/share/vim/vim91)
fchdir() to previous dir
line 23: sourcing "/usr/share/vim/vim91/debian.vim"
finished sourcing /usr/share/vim/vim91/debian.vim
continuing in /etc/vim/vimrc
finished sourcing /etc/vim/vimrc
chdir(/root)
fchdir() to previous dir
could not source "$HOME/.vimrc"
chdir(/root/.vim)
fchdir() to previous dir
could not source "~/.vim/vimrc"
chdir(/root)
fchdir() to previous dir
could not source "$HOME/.exrc"
chdir(/usr/share/vim/vim91)
fchdir() to previous dir
could not source "$VIMRUNTIME/defaults.vim"
Press ENTER or type command to continue
```

Note the last line `could not source "$VIMRUNTIME/defaults.vim"` from which I got the idea of just setting `$VIMRUNTIME` to a custom path.

So if we set `--clean` we can enforce not loading any configs but interestingly `defaults.vim` is still being probed:

```
root@micro01:~# vim.tiny -V -Z --clean /tmp/test
chdir(/tmp)
fchdir() to previous dir
chdir(/usr/share/vim/vim91)
fchdir() to previous dir
could not source "$VIMRUNTIME/defaults.vim"
E1187: Failed to source defaults.vim
Press ENTER or type command to continue
```

Reply:
> I made some tests based on what I found in [canonical/lxd-pkg-snap#237](https://github.com/canonical/lxd-pkg-snap/pull/237). When you don't set `--clean`, then VIM tries to go through the entire chain of directories to find a config file. 

Maybe that's was it was either using `VIMINIT` or `--clean`?

>There is `-V` which gives you a nice rundown of probed files/dirs:

Oh nice, could be handy!

Reply:
> Maybe that's was it was either using VIMINIT or --clean?

Even when setting `$VIMINT=""` it goes through the chain of locations to find any config file. So setting `--clean` brings us the farthest and it skips whatever you have set in `$VIMINT`.

Reply:
Have you tried `VIMINIT` pointing to the file you wanted it to source without looking for it?

Reply:
`/etc/vim/vimrc` doesn't exist in this example :) 
And IIRC `/usr/share/vim/vim91` is the default of `$VIMRUNTIME`.

Yeah I think we have a somewhat stable solution now and no apparmor DENIED messages.